src/simulator_core/entities/assets/ates_cluster.py

```diff
@@ -0,0 +1,276 @@
+        self.thermal_power_allocation = 0
+        self.mass_flowrate = 0
+        self.solver_asset = ProductionAsset(name=self.name, _id=self.asset_id)
+        # ATES default properties
+        self.aquifer_depth = 300.0
+        self.aquifer_thickness = 45.0
+        self.aquifer_mid_temperature = 17.0
+        self.aquifer_net_to_gross = 1.0
+        self.aquifer_porosity = 0.3
+        self.aquifer_permeability = 10000.0
+        self.aquifer_anisotropy = 4.0
+        self.salinity = 10000.0
+        self.well_casing_size = 13.0
+        self.well_distance = 150.0
+        self.maximum_flow_charge = 200.0
+        self.maximum_flow_discharge = 200.0
+
+        # Output list
+        self.output: list = []
+
+    def _calculate_massflowrate(self) -> None:
+        """Calculate mass flowrate of the asset."""
+        self.mass_flowrate = heat_demand_and_temperature_to_mass_flow(
+            self.thermal_power_allocation, self.temperature_supply, self.temperature_return
+        )
+
+    def _set_solver_asset_setpoint(self) -> None:
+        """Set the setpoint of solver asset."""
+        if self.mass_flowrate > 0:
+            self.solver_asset.supply_temperature = self.temperature_return
+        else:
+            self.solver_asset.supply_temperature = self.temperature_supply
+        self.solver_asset.mass_flow_rate_set_point = self.mass_flowrate  # type: ignore
+
+    def set_setpoints(self, setpoints: Dict) -> None:
+        """Placeholder to set the setpoints of an asset prior to a simulation.
+
+        :param Dict setpoints: The setpoints that should be set for the asset.
+            The keys of the dictionary are the names of the setpoints and the values are the values
+        """
+        # Default keys required
+        necessary_setpoints = {
+            PROPERTY_TEMPERATURE_SUPPLY,
+            PROPERTY_TEMPERATURE_RETURN,
+            PROPERTY_HEAT_DEMAND,
+        }
+        # Dict to set
+        setpoints_set = set(setpoints.keys())
+        # Check if all setpoints are in the setpoints
+        if necessary_setpoints.issubset(setpoints_set):
+            self.thermal_power_allocation = setpoints[PROPERTY_HEAT_DEMAND]
+            self.temperature_return = setpoints[PROPERTY_TEMPERATURE_RETURN]
+            self.temperature_supply = setpoints[PROPERTY_TEMPERATURE_SUPPLY]
+
+            self._calculate_massflowrate()
+            self._run_rosim()
+            self._set_solver_asset_setpoint()
+        else:
+            # Print missing setpoints
+            raise ValueError(
+                f"The setpoints {necessary_setpoints.difference(setpoints_set)} are missing."
+            )
+
+    def add_physical_data(self, esdl_asset: EsdlAssetObject) -> None:
+        """Method to add physical data to the asset.
+
+        :param EsdlAssetObject esdl_asset: The esdl asset object to add the physical data from.
+         :return:
+        """
+        self.aquifer_depth, _ = esdl_asset.get_property(
```

Comment:
When the property us not filled in ESDL you get a zero back, how do you check for this

Reply:
i think if the properties is not set, the property will not appear in ESDL. Thus I  put default value